core/src/test/java/ai/timefold/solver/core/impl/testdata/domain/declarative/concurrent_values/TestdataConcurrentValue.java

```diff
@@ -0,0 +1,272 @@
+    public static final LocalDateTime BASE_START_TIME = LocalDate.of(2025, 1, 1).atTime(LocalTime.of(9, 0));
+    String id;
+
+    @InverseRelationShadowVariable(sourceVariableName = "values")
+    TestdataConcurrentEntity entity;
+
+    @ShadowVariable(supplierName = "serviceReadyTimeUpdater")
+    LocalDateTime serviceReadyTime;
+
+    @ShadowVariable(supplierName = "serviceStartTimeUpdater")
+    LocalDateTime serviceStartTime;
+
+    @ShadowVariable(supplierName = "serviceFinishTimeUpdater")
+    LocalDateTime serviceFinishTime;
+
+    @PreviousElementShadowVariable(sourceVariableName = "values")
+    TestdataConcurrentValue previousValue;
+
+    @NextElementShadowVariable(sourceVariableName = "values")
+    TestdataConcurrentValue nextValue;
+
+    List<TestdataConcurrentValue> concurrentValueGroup;
+
+    @ShadowVariableLooped
+    boolean isInvalid;
+
+    public TestdataConcurrentValue() {
+    }
+
+    public TestdataConcurrentValue(String id) {
+        this.id = id;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public TestdataConcurrentEntity getEntity() {
+        return entity;
+    }
+
+    public void setEntity(TestdataConcurrentEntity entity) {
+        this.entity = entity;
+    }
+
+    public TestdataConcurrentValue getPreviousValue() {
+        return previousValue;
+    }
+
+    public void setPreviousValue(TestdataConcurrentValue previousValue) {
+        this.previousValue = previousValue;
+    }
+
+    public TestdataConcurrentValue getNextValue() {
+        return nextValue;
+    }
+
+    public void setNextValue(TestdataConcurrentValue nextValue) {
+        this.nextValue = nextValue;
+    }
+
+    public LocalDateTime getServiceStartTime() {
+        return serviceStartTime;
+    }
+
+    public void setServiceStartTime(LocalDateTime serviceStartTime) {
+        this.serviceStartTime = serviceStartTime;
+    }
+
+    @ShadowSources(sources = { "previousValue.serviceFinishTime", "entity" })
+    public LocalDateTime serviceReadyTimeUpdater() {
+        if (previousValue != null) {
+            return previousValue.serviceFinishTime.plusMinutes(30L);
+        }
+        if (entity != null) {
+            return BASE_START_TIME;
+        }
+        return null;
+    }
+
+    @ShadowSources(sources = { "serviceReadyTime", "concurrentValueGroup[].serviceReadyTime" })
+    public LocalDateTime serviceStartTimeUpdater() {
+        if (serviceReadyTime == null) {
+            return null;
+        }
+        var startTime = serviceReadyTime;
+        if (concurrentValueGroup != null) {
+            for (var visit : concurrentValueGroup) {
+                if (visit.serviceReadyTime != null && startTime.isBefore(visit.serviceReadyTime)) {
+                    startTime = visit.serviceReadyTime;
+                }
+            }
+        }
+        return startTime;
+    }
+
+    @ShadowSources(sources = { "serviceStartTime" })
```

Comment:
Let's try to do it without the mandatory argument name. With just one argument, this should be possible.